### PR TITLE
Feature/gherkin

### PR DESCRIPTION
**Pull request by blacelle**
This follows the work in https://github.com/diffplug/spotless/pull/928 (thanks @jamietanna).

It is reworked to:

1. be synced with main
2. Rely on gherkin-utils core module
3. Lost ability to customize the indentation through this Gherkin formatter. It should remain available through standard Spotless indentation
4. Added maven integration

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -25,3 +25,13 @@ indent_style = space
 [*.{yml,yaml}]
 indent_style = space
 indent_size = 2
+
+# Prevent unexpected automatic indentation when crafting test-cases
+[/testlib/src/main/resources/**]
```
Comment (blacelle) [inline .editorconfig:30]: @nedtwigg I (again) got issues when copy-pasting from Jupiter actual results to test resoures, as IntelliJ was automatically reformatting my clipboard content.

**Comment by jamietanna** [comment]
Thanks for this, but note that there are some issues upstream with how it treats certain things like whitespace and tables, which is also what blocked my original PR with my library 😅

**Comment by nedtwigg** [comment]
@blacelle Would you like me to merge this now, or do you want to wait for https://github.com/cucumber/gherkin-utils/pull/20?

File: plugin-gradle/README.md [hunk 2]
```diff
@@ -850,7 +852,32 @@ spotless {
...
+  gherkin {
+    target 'src/**/*.feature' // you have to set the target manually
+    simple() // has its own section below
+  }
+}
+```
+
+### simple
+
+Uses a Gherkin pretty-printer that optionally allows configuring the number of spaces that are used to pretty print objects:
```
Comment (nedtwigg) [inline plugin-gradle/README.md:870]: As per other formatters (e.g. [scalafmt](https://github.com/diffplug/spotless/tree/main/plugin-gradle#scalafmt)) link to

- https://github.com/cucumber/gherkin-utils
- https://github.com/cucumber/gherkin-utils/blob/main/CHANGELOG.md

File: plugin-maven/CHANGES.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 ### Changes
 * **POTENTIALLY BREAKING** Drop support for `googleJavaFormat` versions &lt; `1.8`. ([#1630](https://github.com/diffplug/spotless/pull/1630))
 * Bump default `googleJavaFormat` version `1.15.0` -> `1.16.0`. ([#1630](https://github.com/diffplug/spotless/pull/1630))
+* Added support for Gherkin feature files ([1649](https://github.com/diffplug/spotless/issues/1649)).
```
Comment (nedtwigg) [inline plugin-maven/CHANGES.md:9]: `#1649` vs `1649`

File: CHANGES.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 ## [Unreleased]
 ### Added
 * Support configuration of mirrors for P2 repositories in `EquoBasedStepBuilder` ([#1629](https://github.com/diffplug/spotless/issues/1629)).
+* Added formatter for Gherkin feature files ([1649](https://github.com/diffplug/spotless/issues/1649)).
```
Comment (nedtwigg) [inline CHANGES.md:15]: `#1649` vs `1649`

**Review by nedtwigg — CHANGES_REQUESTED** [review]
Instead of `gherkin { simple()`, I would name it `gherkin { gherkinUtils()`.

**Comment by blacelle** [comment]
I would merge at your earlier convenience. Let me implement the requested changes.

**Comment by blacelle** [comment]
This should be OK. Thanks @nedtwigg

**Review by nedtwigg — APPROVED** [review]

**Comment by nedtwigg** [comment]
Published in `plugin-gradle 6.18.0` and `plugin-maven 2.36.0`.

**Pull request merged**